app/controllers/works_controller.rb

```diff
@@ -0,0 +1,76 @@
+class WorksController < ApplicationController
+  def index
+    @works = Work.all
+  end
+
+  def show
+    work_id = params[:id].to_i
+    @work = Work.find_by(id: work_id)
+    if @work.nil?
+      not_found_error_notice
+      return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+
+    if @work.save
+      redirect_to work_path(@work.id)
+      return
+    else
+      not_saved_error_notice('create')
+      render :new
+      return
+    end
+  end
+
+  def edit
+    work_id = params[:id].to_i
+    @work = Work.find_by(id: work_id)
+
+    if @work.nil?
+      not_found_error_notice
+      return
+    end
+  end
+
+  def update
+    @work = Work.find_by(id: params[:id])
```

Comment:
This line is repeated in four different actions here making it a great candidate for a small controller filter.